osu.Game/Beatmaps/Beatmap.cs

```diff
@@ -26,11 +27,26 @@ public double BPMAt(double time)
             return 60000 / BeatLengthAt(time);
         }
 
-        public double BeatLengthAt(double time)
+        public double BPMMultiplierAt(double time)
         {
             ControlPoint overridePoint;
             ControlPoint timingPoint = TimingPointAt(time, out overridePoint);
-            return timingPoint.BeatLength;
+
+            return overridePoint?.VelocityAdjustment ?? 1;
+        }
+
+        public double BeatLengthAt(double time, bool addVelocity = true)

```

Comment:
`multiplyVelocity` if anything. That said, I dislike this method. I'd prefer if you completely left the velocity multiplication out, then added a second helper method which just returns the result of this method and multiplies it by the velocity adjustment.